### PR TITLE
add `prepare_weo_2022_scenario()`

### PR DESCRIPTION
**Pull request by cjyetman**
- work towards https://github.com/RMI-PACTA/workflow.scenario.preparation/issues/9
- precedes https://github.com/RMI-PACTA/workflow.scenario.preparation/pull/25

⚠️ The `raw_data_from_provider/used_in_pacta.scenario_preparation/IEA-EV-dataEV salesCarsProjection-APS.csv` file is malformed and needs to be modified for this to work properly.

no differences
```r
waldo::compare(
  arrange(pacta.scenario.preparation::weo_2022, source, scenario, scenario_geography, sector, technology),
  weo_2022,
  tolerance = 1e-15
)
```

### CODE REVIEW

**Comment by jdhoffa** [comment]
> ⚠️ The `raw_data_from_provider/used_in_pacta.scenario_preparation/IEA-EV-dataEV salesCarsProjection-APS.csv` file is malformed and needs to be modified for this to work properly.

When you get a chance, could you elaborate a bit on this? Are @AlexAxthelm and @Antoine-Lalechere aware?

**Comment by cjyetman** [comment]
> > ⚠️ The `raw_data_from_provider/used_in_pacta.scenario_preparation/IEA-EV-dataEV salesCarsProjection-APS.csv` file is malformed and needs to be modified for this to work properly.
> 
> When you get a chance, could you elaborate a bit on this?

It looks like this (with only the last/"source" column specified on the first two lines)
``` csv
region,category,parameter,mode,powertrain,year,unit,value,source
China,Projection-APS,EV sales,Cars,BEV,2020,sales,930000,https://www.iea.org/reports/global-ev-outlook-2022/executive-summary
China,Projection-APS,EV sales,Cars,PHEV,2020,sales,230000
China,Projection-APS,EV sales,Cars,BEV,2021,sales,2700000
```

> Are @AlexAxthelm and @Antoine-Lalechere aware?

no, but now they do?

**Comment by jdhoffa** [comment]
Great. And so I understand, does "needs to be modified for this to work" mean that you will fill in that value for the rest of the rows in the dataset? 

Or are you saying you are blocked somehow until that raw file is "modified" to work?

**Comment by Antoine-Lalechere** [comment]
I added the column source to keep track of our file (prior they were stored on Azure)
The `IEA-EV-dataEV salesCarsProjection-APS.csv` file was extracted from this source: https://www.iea.org/reports/global-ev-outlook-2022
The 8 first columns were downloaded on that page last year and I added the 9th to keep track of it.
I let you decide if you want to remove the 9th now that storage is more efficient.

**Comment by jdhoffa** [comment]
Thanks @Antoine-Lalechere!

**Comment by cjyetman** [comment]
> Great. And so I understand, does "needs to be modified for this to work" mean that you will fill in that value for the rest of the rows in the dataset?
> 
> Or are you saying you are blocked somehow until that raw file is "modified" to work?

I'm currently using a modified version of that file to test the import that I'm creating. I have the expectation that we will want to permanently modify the "raw" file on Azure.

Additionally, a handful of the the "raw" files that I have been working with are currently locked up inside of the `pactarawdata > scenario_sources > RawScenarioData_FromDropbox.zip`. I have the expectation that those directories will be moved into `pactarawdata > scenario_sources` on Azure.

Both expectations are necessary for all of this stuff to work while pulling directly from Azure, at least as it stands now.

**Comment by jdhoffa** [comment]
Tracked here: 
https://dev.azure.com/RMI-PACTA/2DegreesInvesting/_workitems/edit/10608

File: R/prepare_weo_2022_scenario.R [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+      unit = "Units",
+      region = "Scenario geography",
+      scenario = "Scenario",
+      variable = "Variable"
+    ) %>%
+    dplyr::select(-"Data Source")
+
+  oil_formatted <-
+    fossil_fuel %>%
+    dplyr::filter(.data[["Sector"]] == "Oil") %>% # we need to remove Natural Gas Liquids from Oil so it needs "special treatment"
```
Comment (jdhoffa) [inline R/prepare_weo_2022_scenario.R:198]:
```suggestion
    # we need to remove Natural Gas Liquids from Oil so it needs "special treatment"
    dplyr::filter(.data[["Sector"]] == "Oil") %>%
```

File: R/prepare_weo_2022_scenario.R [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+    #     "source",
+    #     "scenario",
+    #     "scenario_geography",
+    #     "indicator",
+    #     "sector",
+    #     "units",
+    #     "technology",
+    #     "year"
+    #   )
+    # ) %>%
```
Comment (jdhoffa) [inline R/prepare_weo_2022_scenario.R:310]:
```suggestion
```

**Review by jdhoffa — APPROVED** [review]
NIT suggestions but LGTM

File: R/prepare_weo_2022_scenario.R [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+                                      weo_2022_fossil_fuels_raw,
+                                      weo_2022_nze_auto_raw,
+                                      weo_2022_nze_steel_raw,
+                                      weo_2022_sales_aps_auto_raw,
+                                      weo_2022_electric_sales_aps_auto_raw) {
+  fossil_fuel <-
+    weo_2022_fossil_fuels_raw %>%
+    dplyr::filter(.data[["Variable"]] == "Supply")
+
+  # A bit of manual calculations here, done on excel
```
[the remark this answers is not in the document]
Comment (cjyetman) [reply R/prepare_weo_2022_scenario.R:35]: yep

**Review by jdhoffa — APPROVED** [review]
LGTM

**Pull request merged**